### ISSUE TITLE
Is there a vertical alignment option or plans for one?

### ISSUE THREAD

**Issue by tamoyal** [rubocop-hq/rubocop#6259]
I'd like to align my code how [this sublime plugin does it](https://github.com/jarod2d/sublime_valign).  Is there any option for that now or plans to do it?

**Comment by bbatsov** [comment]
There are no specific plans about this, but in general I'm open to supporting something like this. I think some layout cops might be supporting vertical alignment even now. @jonas054 will know better.

**Comment by jonas054** [comment]
@tamoyal There is some alignment functionality already. It probably doesn't support everything that the sublime plugin does, but if you put
```yaml
Layout/ExtraSpacing:
  ForceEqualSignAlignment: true

Layout/AlignHash:
  EnforcedColonStyle: table
```
in your `.rubocop.yml` and run `rubocop --auto-correct`, it changes
```ruby
x = 10.0
y = 20.0
width = 100.0
height = 100.0

rectangle = {
  x: 10.0,
  y: 20.0,
  width: 100.0,
  height: 100.0
}

circle = {
  'x' => 10.0,
  'y' => 20.0,
  'radius' => 100.0
}
```
into
```ruby
x      = 10.0
y      = 20.0
width  = 100.0
height = 100.0

rectangle = {
  x:      10.0,
  y:      20.0,
  width:  100.0,
  height: 100.0
}

circle = {
  'x'      => 10.0,
  'y'      => 20.0,
  'radius' => 100.0
}
```

**Comment by jfelchner** [comment]
For equal sign alignment, this should be fixed in #7211.  If you want to have different groups of alignment, you just have to separate them by an empty line.  Each group's assignments will be aligned separately.
